WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -101,17 +101,25 @@ onPage('competitions#index', function() {
 });
 
 onPage("competitions#show_all_results", function() {
-  /* Show tbody for the given event. */
-  function showResultsFor(eventId) {
+  $('.event-selector input[type="radio"]').on('change', function() {
+    var eventId = $(this).val();
+
+    // Select both radio buttons on top and bottom.
+    $("#radio-" + eventId + "-top").prop("checked", "true");
+    $("#radio-" + eventId + "-bottom").prop("checked", "true");
+
+    // Also update the sidebar.
+    $(".nav_event.active").removeClass("active");
+    $(".nav_event .event-" + eventId).parent().addClass("active");
+
+    // Switch to the new event.
     var $results = $('.one-event');
     $results.hide();
     $results.filter('.event-' + eventId).show();
-  }
 
-  /* Handle events selector for results by event. */
-  $('.event-selector input[type="radio"]').on('change', function() {
-    var eventId = $(this).val();
-    showResultsFor(eventId);
+    // Scroll to the top.
+    document.getElementsByClassName('event-selector')[0].scrollIntoViewIfNeeded();
```

Comment:
According to [the documentation](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoViewIfNeeded) `scrollIntoViewIfNeeded` is not a stable API. Would be better to use `scrollIntoView` or some other alternative if you have one in mind.

Reply:
Good catch!  Done.